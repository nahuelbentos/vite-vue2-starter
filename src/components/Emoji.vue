<script lang="ts" setup>
import { computed, ref } from 'vue'

const props = defineProps({
	count: {
		type: Number,
		required: true,
		default: 0
	}
})
const changes = ref(0)
const showList = ref(false)

const emojis = computed(() => {
	changes.value += 1
	return Array.from({ length: props.count }).fill('ðŸ“®')
})

function onClick() {
	showList.value = !showList.value
}
</script>

<template>
	<div>
		<p>Changes: {{ changes }}</p>
		<button @click="onClick" />
		<p v-if="showList">
			<span v-for="(emoji, index) in emojis" :key="index">{{ emoji }}</span>
		</p>
	</div>
</template>
