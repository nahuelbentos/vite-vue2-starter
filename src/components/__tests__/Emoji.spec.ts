import { mount } from '@vue/test-utils'
import Emoji from '../Emoji.vue'

describe('Emoji Component', () => {
	test('should show when enabled 5 emojis', async () => {
		const count = 5
		const wrapper = mount(Emoji, {
			propsData: {
				count
			}
		})
		const button = wrapper.find('button')
		await button.trigger('click')
		const list = wrapper.findAll('span')
		expect(list.length).toBe(count)
	})
})
