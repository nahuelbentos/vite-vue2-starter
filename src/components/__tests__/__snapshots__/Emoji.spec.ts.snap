// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emoji Component should render 5 emojis 1`] = `
"<div>
  <p>Changes: 0</p>
  <p><span>ğŸ“®</span><span>ğŸ“®</span><span>ğŸ“®</span><span>ğŸ“®</span><span>ğŸ“®</span></p>
</div>"
`;
